addons/web/static/src/core/scroller_service.js

```diff
@@ -41,7 +43,14 @@ export const scrollerService = {
                     } catch (e) {
                         // Invalid selector: not an anchor anyway
                     }
-                    if (matchingEl) {
+                    const triggerEv = new CustomEvent("anchor-link-clicked", {
+                        detail: {
+                            element: matchingEl,
+                            id: href.value,
```

Comment:
Maybe pass the original event as well (it might need to be prevent defaulted as welll)
```suggestion
                            originalEv: ev,
```